actioncable/app/javascript/actioncable/subscriptions.js

```diff
@@ -0,0 +1,94 @@
+// Collection class for creating (and internally managing) channel subscriptions. The only method intended to be triggered by the user
+// us ActionCable.Subscriptions#create, and it should be called through the consumer like so:
+//
+//   @App = {}
+//   App.cable = ActionCable.createConsumer "ws://example.com/accounts/1"
+//   App.appearance = App.cable.subscriptions.create "AppearanceChannel"
+//
+// For more details on how you'd configure an actual channel subscription, see ActionCable.Subscription.
+import { Subscription } from "./subscription"
+
+export class Subscriptions {
+  constructor(consumer) {
+    this.consumer = consumer
+    this.subscriptions = []
+  }
+
+  create(channelName, mixin) {
+    const channel = channelName
+    const params = typeof channel === "object" ? channel : {channel}
+    const subscription = new Subscription(this.consumer, params, mixin)
+    return this.add(subscription)
+  }
+
+  // Private
+
+  add(subscription) {
+    this.subscriptions.push(subscription)
+    this.consumer.ensureActiveConnection()
+    this.notify(subscription, "initialized")
+    this.sendCommand(subscription, "subscribe")
+    return subscription
+  }
+
+  remove(subscription) {
+    this.forget(subscription)
+    if (!this.findAll(subscription.identifier).length) {
+      this.sendCommand(subscription, "unsubscribe")
+    }
+    return subscription
+  }
+
+  reject(identifier) {
+    return (() => {
+      const result = []
+      for (let subscription of Array.from(this.findAll(identifier))) {
```

Comment:
I *think* there are browsers that support `WebSocket`, but not `Array.from()` so we should avoid using it if possible.